crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -555,10 +568,29 @@ where
             } else {
                 state.head_block_hash
             };
-            self.sync.set_pipeline_sync_target(target);
+
+            // we need to first check the buffer for the head and its ancestors
+            let lowest_unknown_hash = self
+                .blockchain
+                .lowest_buffered_ancestor(target)
+                .map(|block| block.parent_hash)
+                .unwrap_or(target);
```

Comment:
I'd like to make this a function so we can properly document the behaviour here:

this returns the hash of the lowest unknown block of the target's chain, starting with the target itself.